### PR TITLE
Integrate new vm common

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: arwen/contexts/blockchain.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func (context *blockchainContext) PopSetActiveState() {
 	}
 
 	prevSnapshot := context.stateStack[stateStackLen-1]
-	context.blockChainHook.RevertToSnapshot(prevSnapshot)
+	_ = context.blockChainHook.RevertToSnapshot(prevSnapshot)
```
Comment (iulianpascalau) [inline arwen/contexts/blockchain.go:317]: maybe log the error?

File: arwen/elrondapi/elrondei.go [hunk 2]
```diff
@@ -1513,17 +1511,14 @@ func upgradeContract(
...
-	callData.Bytes(code).Bytes(codeMetadata)
-
+	callData := arwen.UpgradeFunctionName + "@" + hex.EncodeToString(code) + "@" + hex.EncodeToString(codeMetadata)
 	for _, arg := range data {
-		callData.Bytes(arg)
+		callData += "@" + hex.EncodeToString(arg)
 	}
 
-	runtime.ExecuteAsyncCall(
+	_ = runtime.ExecuteAsyncCall(
```
Comment (iulianpascalau) [inline arwen/elrondapi/elrondei.go:1519]: log the error?

File: arwen/hosttest/execution_builtin_test.go [hunk 3]
```diff
@@ -10,9 +10,8 @@ import (
 	contextmock "github.com/ElrondNetwork/arwen-wasm-vm/v1_3/mock/context"
 	worldmock "github.com/ElrondNetwork/arwen-wasm-vm/v1_3/mock/world"
 	test "github.com/ElrondNetwork/arwen-wasm-vm/v1_3/testcommon"
-	"github.com/ElrondNetwork/elrond-go/core"
-	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
-	"github.com/ElrondNetwork/elrond-go/data/esdt"
+	"github.com/ElrondNetwork/elrond-vm-common"
```
Comment (iulianpascalau) [inline arwen/hosttest/execution_builtin_test.go:13]: package contains snake case named files. Maybe a task to rename these?

File: crypto/signing/key.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
+package signing
```
Comment (iulianpascalau) [inline crypto/signing/key.go:1]: We need to add a TODO here to remove this package when we will have elrond-go-core

File: mock/world/gogoProtoMarshalizer.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package worldmock
```
Comment (iulianpascalau) [inline mock/world/gogoProtoMarshalizer.go:1]: Another TODO to remove this when we will have the elrond-go-core

**Review by iulianpascalau — COMMENTED** [review]
minor stuff :+1:

File: mock/world/gogoProtoMarshalizer.go (hunk 5 above)
Comment (sasurobert) [reply mock/world/gogoProtoMarshalizer.go:1]: done

File: crypto/signing/key.go (hunk 4 above)
Comment (sasurobert) [reply crypto/signing/key.go:1]: done

File: arwen/hosttest/execution_builtin_test.go (hunk 3 above)
Comment (sasurobert) [reply arwen/hosttest/execution_builtin_test.go:13]: added

File: arwen/contexts/blockchain.go (hunk 1 above)
Comment (sasurobert) [reply arwen/contexts/blockchain.go:317]: done

File: arwen/elrondapi/elrondei.go (hunk 2 above)
Comment (sasurobert) [reply arwen/elrondapi/elrondei.go:1519]: done

File: arwen/elrondapi/elrondei.go [hunk 6]
```diff
@@ -1513,17 +1511,14 @@ func upgradeContract(
 	// Set up the async call as if it is not known whether the called SC
 	// is in the same shard with the caller or not. This will be later resolved
 	// by runtime.ExecuteAsyncCall().
-	callData := txDataBuilder.NewBuilder()
-	callData.Func(arwen.UpgradeFunctionName)
-	callData.Bytes(code).Bytes(codeMetadata)
-
+	callData := arwen.UpgradeFunctionName + "@" + hex.EncodeToString(code) + "@" + hex.EncodeToString(codeMetadata)
```
Comment (andrei-marinica) [inline arwen/elrondapi/elrondei.go:1514]: There is a txDataBuilder that is currently used for tests, maybe it is worth extracting and using in production code too. Low prio

**Review by iulianpascalau — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

File: arwen/elrondapi/elrondei.go (hunk 6 above)
Comment (iulianpascalau) [reply arwen/elrondapi/elrondei.go:1514]: I think this txDataBuilder will be moved to the upcoming elrond-go-core repo

**Review by matei-p — APPROVED** [review]

**Pull request merged**